include/yk/hash/range.hpp

```diff
@@ -0,0 +1,24 @@
+#ifndef YK_HASH_RANGE_HPP
+#define YK_HASH_RANGE_HPP
+
+#include "yk/hash/hash_combine.hpp"
+
+#include <ranges>
+#include <utility>
+
+#include <cassert>
+#include <cstddef>
+
+namespace yk {
+
+template <std::ranges::range R>
+[[nodiscard]] constexpr std::size_t hash_range(R&& r) noexcept {
+  assert(!std::ranges::empty(r));
```

Comment:
I think empty ranges can and should be hashed

Reply:
If we want to hash an empty range, we need identity element for `hash_combine`.